scratch_space/README.md

```diff
@@ -6,17 +6,41 @@ A `ScratchSpace` is a thin wrapper around a temporary directory. The
 constructor takes zero arguments, so making one is as simple as doing
 `new ScratchSpace()`.
 
+In general you actually want to wrap a `ScratchSpace` in a `Resource`, so that
+you can re-use the scratch space across build steps in an individual build.
+This is safe to do since you are not allowed to overwrite files within a build.
+
+This should look something like the following:
+
+```
+final myScratchSpaceResource =
```

Comment:
do we want to have this as a top level field within the scratch space package? Do we think most uses would want a communal scratch space, or most uses would want to have their own per builder?

Reply:
I opted not to do this for now just because it is less likely to run into issues if you have full control over your own scratch space. If we encourage all users to share one then one badly behaved builder could bork everything.